pkgs/development/libraries/qt-5/5.15/default.nix

```diff
@@ -311,41 +333,19 @@ let
       ] ++ lib.optional (!stdenv.isDarwin) qtwayland
         ++ lib.optional (stdenv.isDarwin) qtmacextras);
 
-      qmake = callPackage ({ qtbase }: makeSetupHook {
-        name = "qmake-hook";
-        ${if stdenv.buildPlatform == stdenv.hostPlatform
-          then "propagatedBuildInputs"
-          else "depsTargetTargetPropagated"} = [ qtbase.dev ];
-        substitutions = {
-          inherit debug;
-          fix_qmake_libtool = ../hooks/fix-qmake-libtool.sh;
-        };
-      } ../hooks/qmake-hook.sh) { };
+      qmake = callPackage ../modules/qmake.nix { inherit debug; inherit qtPlatformCross; };
 
       wrapQtAppsHook = callPackage ({ makeBinaryWrapper, qtbase, qtwayland }: makeSetupHook {
         name = "wrap-qt5-apps-hook";
-        propagatedBuildInputs = [ qtbase.dev makeBinaryWrapper ]
+        propagatedBuildInputs = [ makeBinaryWrapper ];
+        depsTargetTargetPropagated = [ qtbase.dev ]
           ++ lib.optional stdenv.isLinux qtwayland.dev;
```

Comment:
We have no packages in this repository that use `wrapQtAppsHook` and do not depend on `qtbase`. Should we really do this propagation? Is there a real benefit of having `qtwayland` paths in the wrapper if the package itself does not depend on it?